src/lib/store/useDateStore.ts

```diff
@@ -0,0 +1,32 @@
+import { create } from 'zustand';
+
+type State = {
+  currentDate: Date;
+  targetDate: Date;
+  targetMonthFirstDay: Date;
+};
+
+type Action = {
+  prevDate: () => void;
+  nextDate: () => void;
+};
+
+type DateStore = State & Action;
+
+const useDateStore = create<DateStore>((set) => ({
+  currentDate: new Date(),
+  targetDate: new Date(),
+  targetMonthFirstDay: new Date(new Date().getFullYear(), new Date().getMonth(), 1),
```

Comment:
```suggestion
const INITIAL_DATE = new Date();

const useDateStore = create<DateStore>((set) => ({
  currentDate: INITIAL_DATE,
  targetDate: INITIAL_DATE,
  targetMonthFirstDay: new Date(INITIAL_DATE.getFullYear(), INITIAL_DATE.getMonth(), 1),
```

Date 객체를 여러번 생성하고 있네요.
상수로 Date 객체를 한번만 생성해주고 활용해주면 중복 작업을 피할수 있을것 같아요.

Reply:
네! 중복 코드 삭제하고 리팩토링 하도록 하겠습니다.